daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -763,12 +769,179 @@ final class ElementRuntimeData(
+    null, //defaultBitOrderArg: => BitOrder,
+    None, // optPrimTypeArg: => Option[PrimType],
+    null, // targetNamespaceArg: => NS,
+    NS(namespaceURI), // thisElementsNamespaceArg: => NS,
+    null, // optSimpleTypeRuntimeDataArg: => Option[SimpleTypeRuntimeData],
+    null, // optComplexTypeModelGroupRuntimeDataArg: => Option[ModelGroupRuntimeData],
+    0L, // minOccursArg: => Long,
+    0L, // maxOccursArg: => Long,
+    Nope, // maybeOccursCountKindArg: => Maybe[OccursCountKind],
+    local, // nameArg: => String,
+    null, // targetNamespacePrefixArg: => String,
+    null, // thisElementsNamespacePrefixArg: => String,
+    false, // isHiddenArg: => Boolean,
+    false, // isNillableArg: => Boolean,
+    false, // isArrayArg: => Boolean, // can have more than 1 occurrence
+    false, // isOptionalArg: => Boolean, // can have only 0 or 1 occurrence
+    false, // isRequiredInUnparseInfosetArg: => Boolean, // must have at least 1 occurrence
+    LocalDeclQName(None, local, NS(namespaceURI)), // namedQNameArg: => NamedQName,
+    false, // isRepresentedArg: => Boolean,
+    false, // couldHaveTextArg: => Boolean,
+    0, // alignmentValueInBitsArg: => Int,
+    false, // hasNoSkipRegionsArg: => Boolean,
+    null, // impliedRepresentationArg: => Representation,
+    null, // optIgnoreCaseArg: => Option[YesNo],
+    null, // optDefaultValueArg: => Option[AnyRef],
+    null, // optTruncateSpecifiedLengthStringArg: => Option[Boolean],
+    null, // outputValueCalcExprArg: => Option[CompiledExpression[AnyRef]],
+    Nope, // maybeBinaryFloatRepEvArg: => Maybe[BinaryFloatRepEv],
+    Nope, // maybeByteOrderEvArg: => Maybe[ByteOrderEv],
+    Nope, // maybeFillByteEvArg: => Maybe[FillByteEv],
+    Nope, // maybeCheckByteAndBitOrderEvArg: => Maybe[CheckByteAndBitOrderEv],
+    Nope, // maybeCheckBitOrderAndCharsetEvArg: => Maybe[CheckBitOrderAndCharsetEv],
+    false // isQuasiElementArg: => Boolean
+  ) {
+
+  override def toString() = Misc.getNameFromClass(this) + "(" + this.namedQName.toExtendedSyntax + ")"
+
+  @throws(classOf[java.io.IOException])
+  private def writeObject(out: java.io.ObjectOutputStream): Unit =
+    Assert.usageError("Not for serialization")
+}
+
+/**
+ * Used when unparsing to indicate that a next element event was detected that is
+ * unexpected.
+ */
+final class UnexpectedElementErrorERD(
+  optTRD: Option[TermRuntimeData],
+  local: String,
+  namespaceURI: String,
+  val allPossibleNQNs: Seq[QNameBase])
+  extends ErrorERD(local, namespaceURI) {
+
+  System.err.println("Creating.")
+  /**
+   * Causes unparse error with diagnostic about unexpected element.
+   *
+   * Pass argument true if the context is one where no element was expected
+   * (e.g., because an EndElement was expected.)
+   *
+   * Pass false if the Term's ordinary nextElementResolver list of possibilities
+   * is what was expected.
+   */
+  def toUnparseError(nothingWasExpected: Boolean = false) = {
```

Comment:
This is never called. Tests need to be created that get these messages.

But, those will fail because this is never called. Something is missing. 

Also terminology needs to be made the same with the messages in the Element combinator unparsers, and the PartialNextElementResolvers.